web/src/app/components/bar-chart/bar-chart.utils.ts

```diff
@@ -0,0 +1,183 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {formatDate} from '@angular/common';
+import * as d3 from 'd3';
+
+import {step189_2020} from '../../../proto/step189_2020';
+import {DurationItem, findDuration, findDurationUnit, UNIT_CONVERSION} from '../duration-utils';
+
+/**
+ * Item includes all data used by the single in the bar chart.
+ */
+export interface Item {
+  pushID: string;     // Push ID string
+  state: number;      // Tag of the push end state
+  startTime: string;  // Start time of the push, in `yyyy-MM-dd HH:mm:ss` format
+  duration: number;   // Time between last stage and first non-empty stage
+}
```

Comment:
I think we should keep this inside the `bar-chart.component.ts` and imported it from there.